bark/world/objects/agent.hpp

```diff
@@ -193,6 +202,8 @@ class Agent : public Object {
   uint32_t max_history_length_;
   GoalDefinitionPtr goal_definition_;
   double first_valid_timestamp_;
+
+  ObservedWorldPtr sensed_world_;
```

Comment:
Maybe `sampled_observed_world_` or something like this would be better.
How about the concept that we wanted to have two observed worlds? One on the simulation side and one on the agent's side?

At the moment the one from the simulation is set as the one of the agent as well?

Reply:
see comment above

Reply:
why do we need to store it as a member? what is missing with the current implementation of passing it as an argument of the Plan function?

Reply:
We cannot plot it in the viewer since the observed_world is nowhere persisted.